### DISCUSSION TITLE
Trying to get jpexs decompiler working

### DISCUSSION THREAD

**Discussion by eiglow** [lightspark/lightspark#611] [Q&A] [answered]
Hi, I'm having trouble getting jpexs decompiler working. Even the release doesn't work for me, so I figure I must be doing something wrong.

[Here](https://github.com/lightspark/lightspark/files/5785094/ffdec.log) is the terminal output I get when trying to run the latest release zip's `ffdec.sh`.

Trying to build it from source doesn't work for me either, with `ant build` as instructed. [Here](https://github.com/lightspark/lightspark/files/5785100/ffdec-build.log)'s the terminal output I get for that.

Any ideas?

**Comment by huwdp** [comment]
I'll give it a go on Linux later for you. I use it within Windows, didn't know it can run on Linux

**Comment by huwdp** [comment, marked as answer]
I have the same problem. Workaround can be found in the issue linked [here](https://www.free-decompiler.com/flash/issues/1499-ffdec-fails-to-start-could-not-initialize-class-com-jpexs-decompiler-flash-configuration-configuration) (https://www.free-decompiler.com/flash/issues/1499-ffdec-fails-to-start-could-not-initialize-class-com-jpexs-decompiler-flash-configuration-configuration).

**Reply by eiglow** [reply]
According to [this](https://github.com/jindrapetrik/jpexs-decompiler/releases/tag/version12.0.0) release, there is now Java 9 support.

I just tried that though and I still get errors...
```
./ffdec.sh: line 30: [: too many arguments
./ffdec.sh: line 32: [: too many arguments
./ffdec.sh: line 36: [: too many arguments
./ffdec.sh: line 38: [: too many arguments
./ffdec.sh: line 42: [: too many arguments
./ffdec.sh: line 44: [: too many arguments
./ffdec.sh: line 48: [: too many arguments
Error: Unable to initialize main class com.jpexs.decompiler.flash.gui.Main
Caused by: java.lang.NoClassDefFoundError: com/jpexs/debugger/flash/DebugConnectionListener
```

Maybe I will have to look into containers for this stuff, having old versions of java around on linux distros on bare metal is kind of a pain.
